config/samples/custom-theme/README.md

```diff
@@ -0,0 +1,277 @@
+custom/static/_styles.scss
+custom/static/_variables.scss
+custom/static/bootstrap/
+custom/static/bootstrap/_styles.scss
+custom/static/bootstrap/_variables.scss
+custom/static/bootstrap/components/
+custom/static/bootstrap/components/_dropdowns.scss
+custom/static/bootstrap/components/_forms.scss
+custom/static/bootstrap/components/_navbar.scss
+custom/static/bootstrap/components/_navs.scss
+custom/static/bootstrap/components/_type.scss
+custom/static/fonts/
+...
+```
+
+#### 2. Create the theme configuration file
+
+Create `_11_custom_theme.py` with the following content:
+
+```python
+# Override the CUSTOM_THEME_PATH variable with this settings snippet
+AVAILABLE_THEMES = [('custom', 'Custom Theme', 'themes/custom')]
+```
+
+#### 3. Create a ConfigMap with the required files
+
+```bash
+$ oc create cm horizon-theme --from-file=custom.tar.gz --from-file=_11_custom_theme.py
+```
+
+#### 4. Update the OpenStackControlPlane configuration
+
+Edit the `OpenStackControlPlane` resource and update the horizon section:
+
+```yaml
+kind: OpenStackControlPlane
+spec:
+  ...
+  horizon:
+    enabled: true
+    template:
+      extraMounts:
+      - extraVol:
+        - extraVolType: HorizonTheme
+          mounts:
+          - mountPath: /etc/openstack-dashboard/local_settings.d/_11_custom_theme.py
+            name: horizon-theme
+            readOnly: true
+            subPath: _11_custom_theme.py
+          - mountPath: /etc/openstack-dashboard/theme/custom.tar.gz
+            name: horizon-theme
+            readOnly: true
+            subPath: custom.tar.gz
+          volumes:
+          - configMap:
+              name: horizon-theme
+            name: horizon-theme
+  ...
+```
+
+> **Note:** Set the `extraVolType` parameter to `HorizonTheme` to allow the
+horizon-operator to validate the mountPath provided by the user.
+
+
+### Deploy Multiple custom themes
+
+Deploying multiple custom themes is a variation of the single theme deployment
+procedure. This guide demonstrates how to implement two different themes, but
+the same approach can be generalized for any number of custom themes.
+
+#### Prerequisites
+This guide assumes you have already prepared two different theme tarballs:
+
+- custom.tar.gz - First custom theme
+- custom-alt.tar.gz - Second custom theme
+
+#### 1. Prepare the theme configuration file
+
+Create `_11_custom_theme.py` with multiple themes defined:
+
+```python
+# Override the CUSTOM_THEME_PATH variable with this settings snippet
+AVAILABLE_THEMES = [
+    ('custom', 'Custom Theme', 'themes/custom'),
+    ('custom_alt', 'Custom Alt Theme', 'themes/custom_alt')
+]
+```
+
+#### 2. Create ConfigMap with the all required files
+
+```bash
+$ oc create cm horizon-theme --from-file=custom.tar.gz --from-file=_11_custom_theme.py
+$ oc create cm horizon-theme-alt --from-file=custom_alt.tar.gz
+```
+
+> **Note**: It is convenient to have multiple ConfigMaps due to the size of the
+tarball that might exceed the ConfigMap allowed size.
```

Comment:
1 MiB is pretty small. I haven't really looked into any custom Horizon themes to understand how big these tarballs could be. But do we expect any issues there? 

I guess it's not really a problem for us to validate in the Horizon Operator, since the ConfigMap would just fail to be created if it's over 1Mib in size. Rather than explicitly mentioning the size limitation, we can instead link to the k8s documentation about it:
https://kubernetes.io/docs/concepts/configuration/configmap/#motivation

Reply:
right, but in general for simple customizations a tar.gz (binaryData) can be provided by a ConfigMap. This represents our immediate way to test it w/ extraMounts, but clearly we can put files in an NFS share or anything else that is meaningful. I'll defer to the UI team to test alternative solutions if customers require it, but it would be something we build on top of extraMounts (replacing the ConfigMap object w/ something else). In addition, for CI purposes, I think a simple configmap is sufficient to provide only a few files.